pandas/tests/indexes/test_base.py

```diff
@@ -364,7 +356,7 @@ def test_view_with_args_object_array_raises(self, index):
             msg = "When changing to a larger dtype"
             with pytest.raises(ValueError, match=msg):
                 index.view("i8")
-        elif index.dtype == "string":
+        elif index.dtype == "str" and not index.dtype.storage == "python":
             with pytest.raises(NotImplementedError, match="i8"):
                 index.view("i8")
```

Comment:
This is just testing a different error, right? (just to understand why this has a separate branch)

(ideally we should also make it a TypeError, I assume, maybe can add a TODO note)

Reply:
added a TODO